### DISCUSSION TITLE
B2 object storage

### DISCUSSION THREAD

**Discussion by fluffy-critter** [owncast/owncast#3905] [Q&A]
Hi, I've been trying to get B2 object storage to work, but I'm running into problems with it. I know that B2 was previously unsupported due to lack of CORS but it supports CORS now, although the issue I'm running into is with the upload; in my logs I just get:

```
Giving up uploading data/hls/2/stream.m3u8 to object storage https://s3.us-west-000.backblazeb2.com/
```

and the log shows no detailed information about why it's giving up. How can I better diagnose what's happening here?

I've tried configuring the upload with both the bucket name and the bucket ID, and I think I have my keys set correctly although the nomenclature on the B2 dashboard is a bit different than on the Owncast configuration page so it's possible I've got things wrong there.

Thanks!

**Comment by gabek** [comment]
I would expect `s3.us-west-000.backblazeb2.com` to have your bucket name in it. Maybe double check what the endpoint should be.

**Comment by fluffy-critter** [comment]
I doublechecked the endpoint, and in doing so noticed that I forgot to set the bucket public. It seems to be working now. Thanks!
